compiler/resolution/cullOverReferences.cpp

```diff
@@ -45,13 +45,7 @@ shouldUseByValueFunction(FnSymbol*                     fn,
   bool retval = false;
 
   if (refNecessary(se, defMap, useMap) == false) {
-    if (isString(se->var->type->getValType()) &&
-        fn->hasFlag(FLAG_FN_REF_USES_SETTER) == false) {
-      retval = false;
-
-    } else {
-      retval = true;
-    }
+    retval = true;
   }
 
   return retval;
```

Comment:
Actually if this all works, we can inline + retire shouldUseByValueFunction() altogether, get rid of the forward decl for refNecessary(), and remove most? all? of Mike's comments right before cullOverReferences(). This could be done in a later separate PR.